src/Common/ZooKeeper/IKeeper.h

```diff
@@ -34,6 +34,12 @@ struct ACL
     int32_t permissions;
     String scheme;
     String id;
+
+    bool operator<(const ACL & other) const
+    {
+        return std::tuple(permissions, scheme, id)
+            < std::tuple(other.permissions, other.scheme, other.id);
+    }
```

Comment:
```suggestion
    auto operator<=>(const ACL &) const = default;
```
should work here
https://en.cppreference.com/w/cpp/language/default_comparisons